icgcget/utils.py

```diff
@@ -32,18 +32,54 @@ def flatten_dict(d, parent_key='', sep='_'):
     return dict(items)
 
 
+def file_size(num, suffix='B'):
+    for unit in ['', 'K', 'M', 'G', 'T']:
+        if abs(num) < 1024.0:
+            return ["%3.2f" % num,  "%s%s" % (unit, suffix)]
+        num /= 1024.0
+    return ["%.2f" % num, "%s%s" % ('Yi', suffix)]
+
+
+def get_api_url(map):
+    if os.getenv("ICGCGET_API_URL"):
+       api_url = os.getenv("ICGCGET_API_URL")
+    else:
+        api_url = map["portal_url"] + 'api/v1'
+    return api_url
+
+
 def normalize_keys(obj):
     if type(obj) != dict:
         return obj
     else:
         return {k.replace('.', '_'): normalize_keys(v) for k, v in obj.items()}
 
 
-def make_directory(path):
+def config_parse(filename):
+    config = {}
     try:
-        os.makedirs(path)
-    except OSError as exc:
-        if exc.errno == errno.EEXIST and os.path.isdir(path):
-            pass
-        else:
-            raise
\ No newline at end of file
+        config_text = open(filename, 'r')
+    except IOError:
+
+        print("Config file {} not found".format(filename))
+        return config
+
+    try:
+        config_temp = yaml.safe_load(config_text)
+        config_download = flatten_dict(normalize_keys(config_temp))
+        config = {'download': config_download, 'dryrun': config_download, 'logfile': config_temp['logfile']}
+    except yaml.YAMLError:
+
+        print("Could not read config file {}".format(filename))
+        return {}
+
+    return config
+
+
+def match_repositories(repos, copies):
+    for repository in repos:
+        for copy in copies["fileCopies"]:
+            if repository == copy["repoCode"]:
+                return repository, copy
```

Comment:
Should you ever want to grab more  matching values you can make this function return a generator:

`return ((repository, copy) for repository in repos for copy in copies["fileCopies"] if repository == copy["repoCode"])`

Then you can do this: 
`
match_repositories(foo,bar).next()
`